docs/src/style.md

```diff
@@ -1,15 +1,78 @@
 Style guide and design principles
 =================================
 
+This document outlines JuMP's style guide. It also contains a set or principles
+for users writing JuMP models.
+
 Style guide
 -----------
 
-TODO: A style guide for JuMP, JuMP models and surrounding Julia code.
-Formatting, naming, use of macros, comments, TODOs, docstrings, etc.
+!!! info
+    For developers: if you modify JuMP's source code, please fix the style of
+    the surrounding code while you are there.
+
+### JuMP Source Code
+
+ - 80 character line limit. (Except for long URLs in comments.)
+ - 4 spaces for indentation.
+ - Use `lowercase_with_underscore` for variable and function names.
+ - Types and Modules should be in `CamelCase`.
+ - Use `UPPERCASE` for constants.
+ - No bangs (!) in function names. Use docstrings to indicate what is modified
```

Comment:
I definitely disagree with this. This seems unnecessarily confusing and differs from essentially every single Julia package, and Julia itself. Do we really want to be the one package that disobeys an established convention forever?